src/pyhf/infer/utils.py

```diff
@@ -0,0 +1,220 @@
+                = \int\limits_{q_{\mathrm{obs}}}^{\infty} f\left(q\,\middle|s+b\right)\,dq
+                = 1 - F\left(q_{\mathrm{obs}}(\mu)\,\middle|\mu'\right)
+
+            and 1 minus the background only model hypothesis :math:`p`-value
+
+            .. math::
+
+                \mathrm{CL}_{b} = 1- p_{b}
+                = p\left(q \geq q_{\mathrm{obs}}\middle|b\right)
+                = 1 - \int\limits_{-\infty}^{q_{\mathrm{obs}}} f\left(q\,\middle|b\right)\,dq
+                = 1 - F\left(q_{\mathrm{obs}}(\mu)\,\middle|0\right)
+
+            for signal strength :math:`\mu` and model hypothesis signal strength
+            :math:`\mu'`, where the cumulative density functions
+            :math:`F\left(q(\mu)\,\middle|\mu'\right)` are given by Equations (57)
+            and (65) of :xref:`arXiv:1007.1727` for upper-limit-like test
+            statistic :math:`q \in \{q_{\mu}, \tilde{q}_{\mu}\}`.
+            Only returned when ``return_tail_probs`` is ``True``.
+
+            .. note::
+
+                The definitions of the :math:`\mathrm{CL}_{s+b}` and
+                :math:`\mathrm{CL}_{b}` used are based on profile likelihood
+                ratio test statistics.
+                This procedure is common in the LHC-era, but differs from
+                procedures used in the LEP and Tevatron eras, as briefly
+                discussed in :math:`\S` 3.8 of :xref:`arXiv:1007.1727`.
+
+            - :math:`\mathrm{CL}_{s,\mathrm{exp}}`: The expected :math:`\mathrm{CL}_{s}`
+              value corresponding to the test statistic under the background
+              only hypothesis :math:`\left(\mu=0\right)`.
+              Only returned when ``return_expected`` is ``True``.
+
+            - :math:`\mathrm{CL}_{s,\mathrm{exp}}` band: The set of expected
+              :math:`\mathrm{CL}_{s}` values corresponding to the median
+              significance of variations of the signal strength from the
+              background only hypothesis :math:`\left(\mu=0\right)` at
+              :math:`(-2,-1,0,1,2)\sigma`.
+              That is, the :math:`p`-values that satisfy Equation (89) of
+              :xref:`arXiv:1007.1727`
+
+            .. math::
+
+                \mathrm{band}_{N\sigma} = \mu' + \sigma\,\Phi^{-1}\left(1-\alpha\right) \pm N\sigma
+
+            for :math:`\mu'=0` and :math:`N \in \left\{-2, -1, 0, 1, 2\right\}`.
+            These values define the boundaries of an uncertainty band sometimes
+            referred to as the "Brazil band".
+            Only returned when ``return_expected_set`` is ``True``.
+
+    """
+    init_pars = init_pars or pdf.config.suggested_init()
+    par_bounds = par_bounds or pdf.config.suggested_bounds()
+    fixed_params = fixed_params or pdf.config.suggested_fixed()
+
+    calc = create_calculator(
+        calctype,
+        data,
+        pdf,
+        init_pars,
+        par_bounds,
+        fixed_params,
+        qtilde=qtilde,
+        **kwargs,
```

Comment:
At the moment all the `kwargs` are getting passed through and not checked which is then resulting in errors as the `hypotest` specific `kwargs` are not known to `create_calculator`. Example:

```python
>>> import pyhf
>>> model = pyhf.simplemodels.hepdata_like(signal_data=[12.0, 11.0], bkg_data=[50.0, 52.0], bkg_uncerts=[3.0, 7.0])
>>> data = [51, 48] + model.config.auxdata
>>> test_mu = 1.0
>>> pyhf.infer.hypotest(test_mu, data, model, qtilde=True, return_expected=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/utils.py", line 177, in hypotest
    **kwargs,
  File "/home/feickert/Code/GitHub/pyhf/src/pyhf/infer/utils.py", line 39, in create_calculator
    ](*args, **kwargs)
TypeError: __init__() got an unexpected keyword argument 'return_expected'
```

Reply:
This _should_ now be resolved by adding the following above the `create_calculator` call

```python
    # Get the hypotest return specific kwargs
    return_tail_probs = kwargs.pop("return_tail_probs", None)
    return_expected = kwargs.pop("return_expected", None)
    return_expected_set = kwargs.pop("return_expected_set", None)
```